commit d8b1571312b7f77aeae2b2a7a138bb8edaa4f725
Author: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
Date:   Tue Nov 28 17:17:19 2023 -0800

    drm/xe/huc: HuC authentication via GSC
    
    HuC authentication via GSC is performed by submitting the appropriate
    PXP packet to the GSC FW. This packet can trigger a "pending" reply from
    the FW, so we need to handle that and resubmit. Note that the auth via
    GSC can only be performed if the HuC has already been authenticated by
    the GuC.
    
    Signed-off-by: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
    Cc: Alan Previn <alan.previn.teres.alexis@intel.com>
    Cc: John Harrison <John.C.Harrison@Intel.com>
    Cc: Vivaik Balasubrawmanian <vivaik.balasubrawmanian@intel.com>
    Reviewed-by: Vivaik Balasubrawmanian <vivaik.balasubrawmanian@intel.com>
    Signed-off-by: Rodrigo Vivi <rodrigo.vivi@intel.com>

diff --git a/drivers/gpu/drm/xe/xe_uc_fw.c b/drivers/gpu/drm/xe/xe_uc_fw.c
index 9abae65c6b23..73d6938c921d 100644
--- a/drivers/gpu/drm/xe/xe_uc_fw.c
+++ b/drivers/gpu/drm/xe/xe_uc_fw.c
@@ -528,6 +528,8 @@ static int parse_cpd_header(struct xe_uc_fw *uc_fw, const void *data, size_t siz
 		uc_fw->css_offset = offset;
 	}
 
+	uc_fw->has_gsc_headers = true;
+
 	return 0;
 }